### PR TITLE
fix(serde_v8): no panic on reading large text file

### LINKED ISSUE #15489: Deno panicked when reading 1 GB file
When I was trying to test the performance of file reading, I stumbled across a problem with reading large files.

```
$ time deno run --allow-read read.ts

============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: macos x86_64
Version: 1.24.3
Args: ["deno", "run", "--allow-read", "read.ts"]

thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', serde_v8/ser.rs:429:10
stack backtrace:
   0:        0x10ee50c19 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h8926cb4beb3306db
   1:        0x10e478e8a - core::fmt::write::hc1f18fa3b50c7c9d
   2:        0x10ee33488 - std::io::Write::write_fmt::h6b9f4c342a0c9249
   3:        0x10ee36228 - std::panicking::default_hook::{{closure}}::h45609ee8da44383a
   4:        0x10ee35f31 - std::panicking::default_hook::h147f83b3ed514383
   5:        0x10e40925f - deno::setup_panic_hook::{{closure}}::h13d5a32d73c2659e
   6:        0x10ee36c83 - std::panicking::rust_panic_with_hook::h06323e959c8054b4
   7:        0x10ee50f8f - std::panicking::begin_panic_handler::{{closure}}::h0246d09db197196c
   8:        0x10ee50f29 - std::sys_common::backtrace::__rust_end_short_backtrace::h674a492f79294b55
   9:        0x10ee36645 - _rust_begin_unwind
  10:        0x11012c913 - core::panicking::panic_fmt::haf68904a070dc633
  11:        0x11012c967 - core::panicking::panic::ha65927ba0e4fd76b
  12:        0x10edfac46 - <serde_v8::ser::Serializer as serde::ser::Serializer>::serialize_str::h36b763b5512f9254
  13:        0x10e845eb0 - <extern "C" fn(A0) .> R as v8::support::CFnFrom<F>>::mapping::c_fn::hf8c5f4f8c7705a2e
  14:        0x10f290388 - __ZN2v88internal25FunctionCallbackArguments4CallENS0_15CallHandlerInfoE
  15:        0x10f28ff02 - __ZN2v88internal12_GLOBAL__N_119HandleApiCallHelperILb0EEENS0_11MaybeHandleINS0_6ObjectEEEPNS0_7IsolateENS0_6HandleINS0_10HeapObjectEEESA_NS8_INS0_20FunctionTemplateInfoEEENS8_IS4_EENS0_16BuiltinArgumentsE
  16:        0x10f28f3a7 - __ZN2v88internal21Builtin_HandleApiCallEiPmPNS0_7IsolateE
deno run --allow-read read.ts   0.36s  user 0.50s system 97% cpu 0.887 total
avg shared (code):         0 KB
avg unshared (data/stack): 0 KB
total (sum):               0 KB
max memory:                956132 KB
page faults from disk:     0
other page faults:         239634
```

The read.ts file content consist of
```ts
const data = Deno.readTextFileSync("./large.txt");

console.log(data.length);
```

And the large.txt consist of multiple lines consisting of `1 1 1 1 1 1`, generated by using `{repeat 80000000 echo 1 1 1 1 1 1} > large.txt` in `zsh`. The file size is 916 MB.

### PR DESCRIPTION
**Pull request by GJZwiers**
This PR prevents a `panic!` when a user calls `Deno.readTextFileSync` and the string is larger than the v8 limit `kStringMaxLength`. An error is thrown instead.

Fixes #15489. Draft PR because this also happens with the async `Deno.readTextFile` and I want to fix that too.

### CODE REVIEW

File: cli/tests/unit/read_text_file_test.ts [hunk 1]
```diff
@@ -145,3 +145,24 @@ Deno.test(
...
 );
+
+Deno.test(
+  { permissions: { read: true, write: true } },
+  function throwErrorOnReadBigTextFileSync() {
+    const kStringMaxLengthPlusOne = 536870888 + 1;
+    const bytes = new Uint8Array(kStringMaxLengthPlusOne);
+    const filePath = "cli/tests/testdata/too_big_a_file.txt";
+
+    Deno.writeFileSync(filePath, bytes);
```
Comment (bartlomieju) [inline cli/tests/unit/read_text_file_test.ts:156]: How long does it take to run this test?

Comment (GJZwiers) [reply cli/tests/unit/read_text_file_test.ts:156]: `throwErrorOnReadBigTextFileSync ... ok (935ms)`

**Comment by GJZwiers** [comment]
@bartlomieju There is also a panic in `core/runtime.rs` when using the async `Deno.readTextFile` on a large file. I made a change that seems to fix it on a local terminal, but in the test `assertThrows` is failing, because apparently no error is thrown? I'm not too familiar with this part of the codebase, am I perhaps not sending the error from Rust to JS properly?

**Comment by bartlomieju** [comment]
@GJZwiers this looks good to me, can you fix the failing test?

**Comment by GJZwiers** [comment]
@bartlomieju Something is not working yet in the async test, the error is printed to the terminal, but it seems to cause the test run to fail?

**Comment by bartlomieju** [comment]
@GJZwiers please rebase, I'll review it this week

**Comment by bartlomieju** [comment]
@nayeemrmn can you take a look? It looks like a bug in error reporting in `deno test` - instead of being displayed in relevant file section the error is printed at the very end of the test run.

File: core/runtime.rs [hunk 2]
```diff
@@ -1946,7 +1946,7 @@ impl JsRuntime {
 
       for (promise_id, resp) in results.into_iter() {
         args.push(v8::Integer::new(scope, promise_id).into());
-        args.push(resp.to_v8(scope).unwrap());
+        args.push(resp.to_v8(scope)?);
```
[the remark this answers is not in the document]
Comment (GJZwiers) [reply core/runtime.rs]: Many thanks!

**Review by bartlomieju — APPROVED** [review]
LGTM, nice fix @GJZwiers!

Thank you @nayeemrmn too

**Pull request merged**